mapviz_plugins/src/marker_plugin.cpp

```diff
@@ -203,7 +203,39 @@ namespace mapviz_plugins
         markerData.expire_time = ros::Time::now() + lifetime + ros::Duration(5);
       }
 
-      if (markerData.display_type == visualization_msgs::Marker::CYLINDER ||
+      if (markerData.display_type == visualization_msgs::Marker::ARROW)
+      {
+        StampedPoint point;
+        point.color = markerData.color;
+        point.orientation = orientation;
+
+        if (marker.points.empty())
```

Comment:
If a malformed message with exactly 1 point arrives, I think it might be better to just drop it entirely and log the error rather than try to continue on as thought it was a valid message with 0 points.